src/syslog.rs

```diff
@@ -0,0 +1,146 @@
+//! Interfaces for controlling system log.
+
+use crate::NixPath;
+use crate::Result;
+use std::ffi::OsStr;
+
+/// Logging options of subsequent [`syslog`] calls can be set by calling [`openlog`].
+///
+/// The parameter `ident` is a string that will be prepended to every message. The `logopt`
+/// argument specifies logging options. The `facility` parameter encodes a default facility to be
+/// assigned to all messages that do not have an explicit facility encoded.
+pub fn openlog<P: NixPath + ?Sized>(
+    ident: &P,
+    logopt: LogFlags,
+    facility: Facility,
+) -> Result<()> {
+    ident.with_nix_path(|ident| unsafe {
+        libc::openlog(ident.as_ptr(), logopt.bits(), facility as libc::c_int);
+    })
+}
+
+/// Writes message to the system message logger.
+///
+/// The message is then written to the system console, log files, logged-in users, or forwarded
+/// to other machines as appropriate.
+pub fn syslog<S: AsRef<OsStr> + ?Sized>(
```

Comment:
Can we have an example here? It would be great if it could show how to use Rust's built-in formatter with this function.